packages/mui-utils/src/elementAcceptingRef/elementAcceptingRef.test.tsx

```diff
@@ -20,94 +22,83 @@ describe('elementAcceptingRef', () => {
   });
 
   describe('acceptance when not required', () => {
-    let rootNode: HTMLElement;
-
-    function assertPass(element: any, { shouldMount = true } = {}) {
+    async function assertPass(element: any, { shouldMount = true } = {}) {
```

Comment:
What is asynchronous? 🤔

Reply:
`await waitFor`

Reply:
I see, I got confused thinking `await waitFor` was **outside** of `assertPass` 😅 

`waitFor` is suggested for lazy components, is `React.Suspense` using lazy initialization? I'm not very familiar with its implementation 🤔

Reply:
My understanding is that `React.Suspense` + `waitFor` are only required because of the `"accepts lazy"` test, which renders a `React.lazy` component. Kinda weird we use this approach when only a single test requires it, but then the `assertPass` abstraction falls apart.

Maybe we can simplify `assertPass` and just not use it for `"accepts lazy"`. I tried to leave unnecessary changes out of this PR but it's kinda difficult with the `assertPass` abstraction.

Reply:
This, and the other one, which is very similar: elementTypeAcceptingRef.test.tsx

Reply:
Done https://github.com/mui/material-ui/pull/42780/commits/0925920901df2f68a23c03e8b9b3e3bc06ab033d